iothub/device/src/Authentication/IClientIdentity.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client.Transport;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// Interface for device configurations and common attributes.
+    /// </summary>
+    internal interface IClientIdentity
+    {
+        /// <summary>
+        /// Authentication method associated with this client that uses a shared access signature token
+        /// and allows for token refresh.
+        /// </summary>
+        AuthenticationWithTokenRefresh TokenRefresher { get; }
+
+        /// <summary>
+        /// The IoT Hub this client connects to.
+        /// </summary>
+        string IotHubName { get; }
+
+        /// <summary>
+        /// The device ID associated with this client.
+        /// </summary>
+        string DeviceId { get; }
+
+        /// <summary>
+        /// The module ID associated with this client.
+        /// </summary>
+        string ModuleId { get; }
+
+        /// <summary>
+        /// The host service that this client connects to.
+        /// This can either be the IoT hub name or a gateway service name.
+        /// </summary>
+        string HostName { get; }
+
+        // TODO (abmisr): Consolidate with AmqpCbsAudience
+        /// <summary>
+        /// This currently points to the hostname.
+        /// This needs to store the formatted audience (hostname/devices/deviceId)
+        /// </summary>
+        string Audience { get; }
+
+        /// <summary>
+        /// The shared access key name used to connect the device to the IoT hub service.
+        /// </summary>
+        string SharedAccessKeyName { get; }
+
+        /// <summary>
+        /// The shared access key used to connect to the IoT hub service.
+        /// </summary>
+        string SharedAccessKey { get; }
+
+        /// <summary>
+        /// The shared access signature used to connect to the IoT hub service.
+        /// </summary>
+        string SharedAccessSignature { get; }
```

Comment:
I wonder if we should have a sub-type to hold all 3 of these things, since they go together.

Reply:
That's an idea. We should revisit that when we rework the auth classes.